### DISCUSSION TITLE
build issue: configure issue with --with-libzstd-from-internet --with-libb2-from-internet

### DISCUSSION THREAD

**Discussion by dyadav7** [ccache/ccache#595] [Support]
I am unable to build ccache after cloning it. I have to use --with-libzstd-from-internet --with-libb2-from-internet options of configure
      gcc --version      is gcc (GCC) 5.4.0
     ./configure --with-libzstd-from-internet --with-libb2-from-internet
     make
       gcc -O3 -Wall -Wextra -Wcast-qual -Wcast-align -Wshadow -Wstrict-aliasing=1 -Wswitch-enum -Wdeclaration-after-statement -Wstrict-prototypes -Wundef -Wpointer-arith -Wvla -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings -Wredundant-decls -Wmissing-prototypes -Wc++-compat  -I. -I./common -DXXH_NAMESPACE=ZSTD_ -DZSTD_LEGACY_SUPPORT=0  -c -o compress/huf_compress.o compress/huf_compress.c
/tmp/ccAa8qXc.s: Assembler messages:
/tmp/ccAa8qXc.s:526: Error: no such instruction: `shlx %r14,%rsi,%rsi'
/tmp/ccAa8qXc.s:529: Error: no such instruction: `shlx %rcx,%r14,%rcx'
/tmp/ccAa8qXc.s:535: Error: no such instruction: `shlx %r15,%r9,%r9'
/tmp/ccAa8qXc.s:540: Error: no such instruction: `shlx %r10,%r9,%r9'
/tmp/ccAa8qXc.s:554: Error: no such instruction: `shrx %r11,%rax,%r11'
/tmp/ccAa8qXc.s:559: Error: no such instruction: `shlx %rcx,%rax,%rax'
/tmp/ccAa8qXc.s:615: Error: no such instruction: `shlx %rcx,%rax,%rax'
/tmp/ccAa8qXc.s:623: Error: no such instruction: `shlx %rcx,%rsi,%rsi'
/tmp/ccAa8qXc.s:636: Error: no such instruction: `shrx %r11,%rax,%r11'
make[1]: *** [<builtin>: compress/huf_compress.o] Error 1

If i used 32bit gcc then make works fine,

**Comment by jrosdahl** [comment]
I don't see any inline assembler that looks like that in zstd's source code, so it looks like your compiler creates assembler instructions that your assembler doesn't support. Have you built GCC yourself or does it come from some Linux distribution? If so, which one?

Are you able to build zstd standalone (`git clone https://github.com/facebook/zstd.git && cd zstd && make`)?

**Comment by dyadav7** [comment]
i did a "make clean "and i was able to compile ccache today.
It looks if autogen.sh and configure is done without make clean then it does not re-compile everything.
